### ISSUE TITLE
unit conversion becomes dimensionless

### ISSUE THREAD

**Issue by rcavazz** [astropy/astropy#16704]
### Description

is this correct?
```
>>> import astropy.units as au
>>> (au.cm**3).to( au.liter )
0.001
>>> (1 * au.cm**3).to( au.liter )
<Quantity 0.001 l>
>>>
```

### Expected behavior

I would expect something like:
```
>>> import astropy.units as au
>>> (au.cm**3).to( au.liter )
<Quantity 0.001 l>
>>> (1 * au.cm**3).to( au.liter )
<Quantity 0.001 l>
>>>
```

### How to Reproduce

1. Get package from '...'
2. Then run '...'
3. An error occurs.

```python
# Put your Python code snippet here.
```

### Versions

6.1.1

**Comment by pllim** [comment]
I think the behavior is intentional (I vaguely remember this had been brought up in a different issue before but not sure how to dig it up).

`au.cm**3` is a *unit*, while ` (1 * au.cm**3)` is a *Quantity*.

**Comment by rcavazz** [comment]
thank you for your answer. 
The definition of the quantity was a quick shortcut to avoid some obvious problems. 
The final solution I made is to explicitly define a new unit with `astropy.unit.Unit()` method.
However to be consistent with the package behavior, a combination of units (unless dimensionless), should return another unit, as happens during the implicit quantity creation.

**Comment by pllim** [comment]
I vaguely remember that changing `unit.to` behavior would break a lot of stuff related to conversion machinery but let's ask... @mhvk or @nstarman , do you remember more details about this or the issue?

**Comment by mhvk** [comment]
I don't think we can change the behaviour such that the example on top would become consistent: `Unit.to` is really for converting a *value* (default: 1) in some units to another value, while `Quantity.to` converts a quantity to a new one, with a different unit. In hindsight, `Unit.to` is arguably a bad method name, but that is far too late to change now.

**Comment by rcavazz** [comment]
ok.
to me it could be fine, but I think this drawback should reported in the documentation, at least in the api one:
https://github.com/astropy/astropy/blob/main/astropy/units/core.py#L1193
where it says:
`Return the converted values in the specified unit.`
but in practice behaves differently for units and quantities. e.g.
```
>>> import astropy.units as au
>>> au.imperial.gallon.to( au.m**3 )
0.0037854117891320325
>>> CFM = au.Unit( "CFM", au.imperial.ft**3 / au.min )
>>> CFM.to( au.m**3 / au.s ), (1* CFM).to( au.m**3 / au.s )
(0.00047194744320000004, <Quantity 0.00047195 m3 / s>)
```

**Comment by pllim** [comment]
Minor doc update to make things clearer is welcome! The return does explain a bit more though.

https://github.com/astropy/astropy/blob/9b351c9ead47554be5b1e1eae49124be5d3c3427/astropy/units/core.py#L1213-L1217
